src/sentry/static/sentry/app/actionCreators/events.tsx

```diff
@@ -77,3 +81,76 @@ export const doEventsRequest = (
     },
   });
 };
+
+export type EventQuery = {
+  field: string[];
+  project?: string | string[];
+  sort?: string | string[];
+  query: string;
+  per_page?: number;
+  referrer?: string;
+};
+
+export type TagSegment = {
+  count: number;
+  name: string;
+  value: string;
+  url: LocationDescriptor;
+  isOther?: boolean;
+  key?: string;
+};
+
+export type Tag = {
+  key: string;
+  topValues: Array<TagSegment>;
+};
+
+/**
+ * Fetches tag facets for a query
+ *
+ * @param {Object} api
+ * @param {String} orgSlug
+ * @param {String} query
+ * @returns {Promise<Object>}
+ */
+export function fetchTagFacets(
+  api: Client,
+  orgSlug: string,
+  query: EventQuery
+): Promise<Tag[]> {
+  const urlParams = pick(query, Object.values(URL_PARAM));
+
+  const queryOption = {...urlParams, query: query.query};
+
+  return api.requestPromise(`/organizations/${orgSlug}/events-facets/`, {
+    query: queryOption,
+  });
+}
+
+/**
+ * Fetches total count of events for a given query
+ *
+ * @param {Object} api
+ * @param {String} orgSlug
+ * @param {string} query
+ * @returns {Promise<Number>}
+ */
+export function fetchTotalCount(
+  api: Client,
+  orgSlug: String,
+  query: EventQuery
+): Promise<number> {
+  const urlParams = pick(query, Object.values(URL_PARAM));
+
+  const queryOption = {...urlParams, query: query.query};
+
+  type Response = {
+    count: number;
+  };
+
+  return api
```

Comment:
Why not async/await?

Reply:
Was mostly forklifting the code around. I can switch it around though.